### PR TITLE
Tweak the conversion script to be independent of naming conventions

### PR DESCRIPTION
**Pull request by jwsmithers**
In the past we assumed a set path for layer names. But if someone adds a "/"
then this breaks the conversion. Now we do a recursive search through each
level of the HDF5 path to see if it's a Group or Dataset. It stops and
returns the dataset.

### CODE REVIEW

**Comment by jwsmithers** [comment]
Solves #72 which turned out to not really be what the issue was about.

**Comment by dguest** [comment]
Cool thanks for fixing this josh. It seems a bit weird that we need to do a recursive search of groups to find matches (and I worry about how this could break): shuldn't there be a unique path determined by the architecture file?

Anyway, I guess for now we focus on something that works and start thinking about how to clean up the converters in general.

**Comment by jwsmithers** [comment]
I couldn't think of a foolproof way to always get the path to the dataset. As Andreas's example showed there might be some unknown string attached to the path that doesn't show up in the "name" parameter of the config.

The only immediate way I can see this breaking is if in some hdf5 config file you have a group mixed in with a dataset. I.e.:
```
batch_normalization_2
gamma
beta
```

But this would be downright weird and I don't _think_ hdf5 supports this.

The advantage of this method is that we really don't care how people name the layers, or even how tensorflow vs theano vs sklearn will save the parameters which is quite useful.

**Comment by dguest** [comment]
hdf5 lets you mix groups and datasets, but I agree that I haven't seen the structure you're talking about. What I don't understand is how Keras finds the datasets when it loads them. I tried to figure this out based on the topology module: 

https://github.com/fchollet/keras/blob/master/keras/engine/topology.py

but it's not exactly clear from a quick glance
